File: core/src/net/brenig/pixelescape/render/overlay/OverlayWithUi.java
package net.brenig.pixelescape.render.overlay;

import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import net.brenig.pixelescape.lib.Reference;
import net.brenig.pixelescape.render.ui.ingame.StageManagerGame;
import net.brenig.pixelescape.screen.GameScreen;

/**
 *
 */
public abstract class OverlayWithUi extends Overlay {

	protected final StageManagerGame stage;

	public OverlayWithUi(GameScreen screen) {
		super(screen);
		stage = new StageManagerGame(screen);
	}

	@Override
	public void render(float delta) {
		screen.getGame().getFont().getData().setScale(Reference.GAME_UI_MAIN_MENU_FONT_SIZE);
		stage.act(delta);
		stage.draw(screen.getGame().getRenderManager());
	}

	@Override
	public void show() {
		screen.setOverlayInputProcessor(stage.getInputProcessor());
	}

	@Override
	public void onResize(int width, int height) {
		stage.updateStageToGameBounds(width, height);
		super.onResize(width, height);
	}

	@Override
	public void dispose() {
		stage.dispose();
	}

	@Override
	public boolean shouldHideGameUI() {
		return true;
	}

	protected Skin getSkin() {
		return screen.getGame().getSkin();
	}
}

File: core/src/net/brenig/pixelescape/render/overlay/OverlayWithUi.kt
package net.brenig.pixelescape.render.overlay

import com.badlogic.gdx.scenes.scene2d.ui.Skin
import net.brenig.pixelescape.lib.Reference
import net.brenig.pixelescape.render.ui.ingame.StageManagerGame
import net.brenig.pixelescape.screen.GameScreen

/**
 *
 */
abstract class OverlayWithUi(screen: GameScreen) : Overlay(screen) {

    protected val stage: StageManagerGame = StageManagerGame(screen)

    protected val skin: Skin
        get() = screen.game.skin

    override fun render(delta: Float) {
        screen.game.font.data.setScale(Reference.GAME_UI_MAIN_MENU_FONT_SIZE)
        stage.act(delta)
        stage.draw(screen.game.renderManager)
    }

    override fun show() {
        screen.setOverlayInputProcessor(stage.inputProcessor)
    }

    override fun onResize(width: Int, height: Int) {
        stage.updateStageToGameBounds(width, height)
        super.onResize(width, height)
    }

    override fun dispose() {
        stage.dispose()
    }

    override fun shouldHideGameUI(): Boolean {
        return true
    }
}
